apps/game_backend/lib/game_backend/items.ex

```diff
@@ -244,4 +244,35 @@ defmodule GameBackend.Items do
       {:error, :character_cannot_equip}
     end
   end
+
+  @doc """
+  Gets Item's purchase cost by given currency and item_template.
+  Returns {:ok, purchase_cost} if found one.
+  Returns {:error, :not_found} if there are none.
+  Fails if there are more than one.
+  """
+  def get_purchase_cost_by_currency(currency_id, item_template) do
+    purchase_cost =
+      Map.get(item_template, :purchase_costs)
+      |> Enum.find(fn purchase_cost -> purchase_cost.currency_id == currency_id end)
+
+    case purchase_cost do
+      nil -> {:error, :not_found}
+      purchase_cost -> {:ok, purchase_cost}
+    end
+  end
+
+  @doc """
+  Receives a user_id, an item_template_id and a list of CurrencyCosts.
+  Inserts new Item from given ItemTemplate for given User.
+  Substract the amount of Currency to User by given params.
+  Returns {:ok, map_of_ran_operations} in case of success.
+  Returns {:error, failed_operation, failed_value, changes_so_far} if one of the operations fail.
+  """
+  def buy_item(user_id, template_id, purchase_costs_list) do
+    Multi.new()
+    |> Multi.run(:item, fn _, _ -> insert_item(%{user_id: user_id, template_id: template_id}) end)
```

Comment:
```suggestion
    |> Multi.run(:insert_item, fn _, _ -> insert_item(%{user_id: user_id, template_id: template_id}) end)
```

Reply:
Ehhhh I like :item more because in the resulting map you'll have something like %{item: %Item{}}.
We can talk about it or I can change it nevertheless if you disagree.

Reply:
Hmmm I think it's kind of the same, but using the generic name :item you don't know exactly what's happening to that item inside the multi, it's not a mandatory change, you can omit this.

Reply:
Ok. I'll skip it then we can discuss it tomorrow at office, so we standarize this.

Reply:
I also like `:item` more and have been using it in the Autobattler and GameBackend apps! Although I agree `:insert_item` is more descriptive, I feel like the insert is clear enough in the code to not need that explicitness :P